File: 205-isomorphic-strings/205.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        table_s_t = {}
        table_t_s = {}
        i = 0

        while i < len(s):
            if t[i] in table_t_s and table_t_s[t[i]] != s[i]:
                return False
            table_s_t[s[i]] = t[i]
            table_t_s[t[i]] = s[i]
            i += 1

        new_str = ""

        i = 0
        while i < len(s):
            new_str += table_s_t[s[i]]
            i += 1

        return new_str == t

File: 205-isomorphic-strings/205.cpp
class Solution {
private:
    bool _isIsomorphic(string s, string t) {
        unordered_map<char, char> m;
        for (int i = 0; i < s.size(); ++i) {
            char s0 = s[i];
            char t0 = t[i];
            if (m.find(s0) == m.end()) {
                m[s0] = t0;
            } else {
                if (m[s0] != t0) {
                    return false;
                }
            }
        }
        return true;
    }
public:
    bool isIsomorphic(string s, string t) {
        return _isIsomorphic(s, t) && _isIsomorphic(t, s);
    }
};
